modules/standard/Memory/Initialization.chpl

```diff
@@ -98,6 +101,47 @@ module Initialization {
     }
   }
 
+  /*
+    Move-initialize ``dst`` with the value in ``src``. The contents of ``dst``
+    are not deinitialized before the move, and ``src`` is not deinitialized
+    after the move.
+
+    .. warning::
+
+      If ``dst`` references an already initialized variable, it will be
+      overwritten by the contents of ``src`` without being deinitialized
+      first. Call :proc:`explicitDeinit()` to deinitialize ``dst`` if
+      necessary.
+
+    .. warning::
+
+      The static types of ``dst`` and ``src`` must match, or else a
+      compile-time error will be issued.
+
+    .. note::
+
+      If the compiler inserts a copy for the argument to ``src``, then a
+      compile-time error will be issued. The most likely cause is that the
+      argument is used elsewhere following the call to ``moveInitialize``.
+      The ``moveFrom`` function can be used with the ``src`` argument to avoid
+      the copy when certain of the variable's usage:
+
+      moveInitialize(myDst, moveFrom(mySrc));
+
+    :arg dst: A variable to move-initialize, whose type matches ``src``
+
+    :arg src: A value to move-initialize from
+  */
+  proc moveInitialize(ref dst,
```

Comment:
Similarly, since we're replacing this module with a new one, I'd have the deprecation message point to the one in MemMove instead of adding a new function that's kinda already deprecated.

Reply:
In that case should I add deprecation warnings to the other functions? Or were you thinking of something more specific like:
```chpl
deprecated "The formals 'lhs' and 'rhs' are deprecated, please use the version in the 'MemMove' module with 'dst' and 'src' instead"
```

Deprecating these things across both modules is a bit weird, but I was trying to avoid the case where a user encountered one error, fixed it, then encountered another they weren't told about previously.

Reply:
I was thinking a more specific deprecation message that pointed straight to the new function that won't give them any deprecation warnings at all.  It'll be a little annoying if they try to silence this error, then still have the module one, while pointing them at the new module might enable them to make the changes they need all at once

Reply:
Offline Lydia suggested no-docing the deprecated functions in the ``MemMove`` module as a compromise.